### PR TITLE
[ci] Updating lmi-dist ci tests for rubikon-engine

### PR DESCRIPTION
**Pull request by maaquib**
## Description ##

Following changes have been made
* Removes bitsandbytes test case, since we no longer support it
* Adds speculative decoding test case
* Adds tests for Gemma-7B and Santacoder

- If this change is a backward incompatible change, why must this change be made? - We're dropping support for bitsandbytes quantization
- Interesting edge cases to note here - N/A

### CODE REVIEW

File: .github/workflows/rolling_batch_integration.yml [hunk 1]
```diff
@@ -299,14 +299,34 @@ jobs:
             serve -m test=file:/opt/ml/model/test/
           python3 llm/client.py lmi_dist octocoder
           docker rm -f $(docker ps -aq)
-      - name: Test gpt-neox-20b-bitsandbytes
+      - name: Test speculative-llama-13b
         working-directory: tests/integration
         run: |
           rm -rf models
-          python3 llm/prepare.py lmi_dist gpt-neox-20b-bitsandbytes
+          echo -en "SPECULATIVE=true" > docker_env
```
Comment (lanking520) [inline .github/workflows/rolling_batch_integration.yml:306]: what is this for? This is new to me

**Review by lanking520 — COMMENTED** [review]
could you also remove the speculative decoding test in vLLM?

File: .github/workflows/rolling_batch_integration.yml (hunk 1 above)
Comment (maaquib) [reply .github/workflows/rolling_batch_integration.yml:306]: In the current implementation the patch is applied when this env variable is set. @chen3933 to confirm

**Review by lanking520 — APPROVED** [review]

File: serving/docker/deepspeed.Dockerfile [hunk 2]
```diff
@@ -20,7 +20,7 @@ ARG torch_vision_version=0.16.2
 ARG pydantic_version=2.6.1
 # HF Deps
 ARG protobuf_version=3.20.3
-ARG transformers_version=4.38.1
+ARG transformers_version=4.39.0
 ARG accelerate_version=0.27.2
```
Comment (lanking520) [inline serving/docker/deepspeed.Dockerfile:24]: 0.28.0

**Comment by maaquib** [comment]
Test run: https://github.com/deepjavalibrary/djl-serving/actions/runs/8382943920

**Review by davidthomas426 — APPROVED** [review]
Is it ok for our CI tests to use HF model ids directly rather than s3 links?

Other than that, LGTM.

**Comment by lanking520** [comment]
> Is it ok for our CI tests to use HF model ids directly rather than s3 links?
> 
> Other than that, LGTM.

There is a problem with HF id. Firstly it is not very stable. Secondly, sometimes the download speed is very long and cause timeout (flaky tests)

**Comment by lanking520** [comment]
@maaquib could you also just replace the lmi-dist handler with the new one in this PR, so we can just cover in CI testing?

**Comment by maaquib** [comment]
> could you also add tests for vLLM as well, just to serve as baseline?

Added gemma and starcoder2 to vllm

**Review by rohithkrn — APPROVED** [review]

File: tests/integration/llm/client.py [hunk 3]
```diff
@@ -373,7 +384,15 @@ def get_model_name():
         "batch_size": [1, 4],
         "seq_length": [256],
         "tokenizer": "mistralai/Mistral-7B-v0.1"
-    }
+    },
+    "llama-7b-unmerged-lora": {
```
[the remark this answers is not in the document]
Comment (maaquib) [reply tests/integration/llm/client.py:388]: It's should not be blocking. I have not added it to the `.github/workflows/llm_integration.yml` yet. We can do that once the lora plumbing in the engine is done.

**Comment by maaquib** [comment]
Build: https://github.com/deepjavalibrary/djl-serving/actions/runs/8395244232
